packages/omgidl-parser/src/IDLNodes/UnionIDLNode.ts

```diff
@@ -0,0 +1,86 @@
+import { IDLNode } from "./IDLNode";
+import { StructMemberIDLNode } from "./StructMemberIDLNode";
+import { AnyIDLNode, IUnionIDLNode } from "./interfaces";
+import { UnionASTNode } from "../astTypes";
+import { INTEGER_TYPES, SIMPLE_TYPES } from "../primitiveTypes";
+import { Case, IDLMessageDefinition, IDLMessageDefinitionField } from "../types";
+
+export class UnionIDLNode extends IDLNode<UnionASTNode> implements IUnionIDLNode {
+  constructor(scopePath: string[], astNode: UnionASTNode, idlMap: Map<string, AnyIDLNode>) {
+    super(scopePath, astNode, idlMap);
+  }
+
+  get type(): string {
+    return this.astNode.name;
+  }
+
+  get isComplex(): boolean {
```

Comment:
Maybe an unpopular opinion but I don't like this "isComplex" attribute name. It's something we inherited from the old days of webviz ROS message support. The name doesn't seem to do a great job of describing what it means (since we don't really use the word "complex" elsewhere in our code so it's not really a well-known term). Maybe someday we can come up with a better name or a different way of representing this information.

Reply:
Yeah I definitely agree. It's really not evident how this is going to be used or what it means. Right now I'm just using it to signify that these types are not _resolved_ like non-complex types and don't have direct, simple serializers. I'm using the `aggregatedKind` attribute (taken from the IDL/DDS spec) to augment this, but maybe there's a variable name that's better for `MessageDefinition`? i am trying to keep this mostly compliant with `MessageDefinition` for now. Food for thought for the future though